e2e/libs/csi_volume_snapshot/csi_volume_snapshot.py

```diff
@@ -184,3 +185,41 @@ def cleanup_csi_volume_snapshots(self):
                     assert delete_exception.status == 404
         except ApiException as list_exception:
             assert list_exception.status == 404
+
+    def remove_csi_volume_snapshot_finalizer(self, csi_volume_snapshot_name):
+        for i in range(self.retry_count):
+            logging(f"Removing finalizers from CSI volume snapshot {csi_volume_snapshot_name} ... ({i})")
+            try:
+                snapshot = self.api.get_namespaced_custom_object(
+                    group=self.group,
+                    version=self.version,
+                    namespace="default",
+                    plural="volumesnapshots",
+                    name=csi_volume_snapshot_name
+                )
+                if not snapshot.get("metadata", {}).get("finalizers"):
+                    return
+
+                snapshot["metadata"]["finalizers"] = []
+                self.api.replace_namespaced_custom_object(
+                    group=self.group,
+                    version=self.version,
+                    namespace="default",
+                    plural="volumesnapshots",
+                    name=csi_volume_snapshot_name,
+                    body=snapshot
+                )
+                return
+            except ApiException as e:
+                if e.status == 404:
+                    return
+            time.sleep(self.retry_interval)
+        assert False, f"Failed to remove finalizers from csi volume snapshot {csi_volume_snapshot_name}"
+
+    async def delete_then_remove_finalizer(self, snapshot_name):
```

Comment:
@chriscchien Could you help try 
```
kubectl patch volumesnapshot test-csi-volume-snapshot --type=merge -p '{"metadata":{"finalizers":[]}}' | kubectl delete VolumeSnapshot test-csi-volume-snapshot
```
to merge 2 operations `removing finalizers` and  `deleting the VolumeSnapshot` into single command? If it works, we can omit the complicated `async/await` logic.

Reply:
Updated.

Using a pipe in a single command works because the delete operation is executed immediately after the patch. This happens before the CSI snapshotter re-add the finalizer, so the VolumeSnapshot deletion succeeds.